gamemodes/terrortown/gamemode/shared/sh_marker_vision_object.lua

```diff
@@ -0,0 +1,244 @@
+-- @return Color The color
+-- @realm shared
+function MARKER_VISION_OBJECT:GetColor()
+	return self.data.color
+end
+
+---
+-- Sets the marked entity to the element.
+-- @param Entity ent The marked entity
+-- @realm shared
+function MARKER_VISION_OBJECT:SetEnt(ent)
+	self.data.ent = ent
+end
+
+---
+-- Gets the marked entity to the element.
+-- @return Entity The marked entity
+-- @realm shared
+function MARKER_VISION_OBJECT:GetEnt()
+	return self.data.ent
+end
+
+---
+-- Sets the unique identifier to the element.
+-- @param string identifier The unique identifier
+-- @realm shared
+function MARKER_VISION_OBJECT:SetIdentifier(identifier)
+	self.data.identifier = identifier
+end
+
+---
+-- Gets the unique identifier to the element.
+-- @return string The unique identifier
+-- @realm shared
+function MARKER_VISION_OBJECT:GetIdentifier()
+	return self.data.identifier
+end
+
+---
+-- Sets the mark owner to the element.
+-- @param Entity owner The mark owner
+-- @realm shared
+function MARKER_VISION_OBJECT:SetOwner(owner)
+	self.data.owner = owner
+end
+
+---
+-- Gets the mark owner to the element.
+-- @return Entity The mark owner
+-- @realm shared
+function MARKER_VISION_OBJECT:GetOwner()
+	return self.data.owner
+end
+
+---
+-- Sets the visible for flag to the element.
+-- @param number visibleFor The visible for flag
+-- @realm shared
+function MARKER_VISION_OBJECT:SetVisibleFor(visibleFor)
+	self.data.visibleFor = visibleFor
+end
+
+---
+-- Gets the visible for flag to the element.
+-- @return number The visible for flag
+-- @realm shared
+function MARKER_VISION_OBJECT:GetVisibleFor()
+	return self.data.visibleFor
+end
+
+---
+-- Gets the visible for flag as a language string to the element.
+-- @return string The visible for flag as a language string
+-- @realm shared
+function MARKER_VISION_OBJECT:GetVisibleForString()
+	return "marker_vision_visible_for_" .. tostring(self:GetVisibleFor())
+end
+
+---
+-- Checks if this element is the searched object defined by the unique
+-- identifier and the entity that is marked.
+-- @param Entity ent The entitiy that is possibly marked
+-- @param string identifier The unique identifier
+-- @return boolean Returns true if the element matches
+-- @realm shared
+function MARKER_VISION_OBJECT:IsSearchedObject(ent, identifier)
+	return self.data.ent == ent and self.data.identifier == identifier
+end
+
+---
+-- Updates the relations of this marker vision element.
+-- @note This does not sync it to the client, call @{MARKER_VISION_OBJECT:SyncToClients} to sync to client.
+-- @param number|string|Player owner The owner of the wallhack that takes their ownership with them on
+-- team change; can also be a team (string) or role (number) if it shouldn't be bound to a player
+-- @param number visibleFor Visibility setting: `VISIBLE_FOR_PLAYER`, `VISIBLE_FOR_ROLE`, `VISIBLE_FOR_TEAM`,
+-- `VISIBLE_FOR_ALL`
+-- @realm shared
+function MARKER_VISION_OBJECT:UpdateRelations(owner, visibleFor)
```

Comment:
But these two alre always set together. Not sure if splitting them up is actually helpful?

Reply:
It would certainly be much cleaner and make the naming easier.
Sometimes or rather most of the time the owner is the only thing that dynamically changes.